lib/protocol/bep_extensions.go

```diff
@@ -145,7 +153,14 @@ func (f FileInfo) IsEmpty() bool {
 //  - target
 // A directory does not have anything specific to check.
 func (f FileInfo) IsEquivalent(other FileInfo, ignorePerms bool, ignoreBlocks bool) bool {
-	if f.Name != other.Name || f.Type != other.Type || f.Deleted != other.Deleted || f.Invalid != other.Invalid {
+	if f.MustRescan() || other.MustRescan() {
+		// These are per definition not equivalent because they don't
+		// represent a valid state, even if both happen to have the
+		// MustRescan bit set.
+		return false
+	}
+
+	if f.Name != other.Name || f.Type != other.Type || f.Deleted != other.Deleted || f.Invalid != other.Invalid || f.LocalFlags != other.LocalFlags {
```

Comment:
So it also checks `LocalFlags` on both fileinfos. I think `IsEquivalent` is really only used by the walker, so it won't have to compare local against remote files, but if it did it would (correctly, I think) consider them non-equivalent.

Reply:
That's not what I intended `IsEquivalent` to be (check if file info is representing the same file, regardless of internal metadata) and it is already used outside of walker, in send-only folders. There it does actually compare the current local file with needed current global files, to prevent out-of-sync items which are actually identical. For example mutually ignored items with conflicting version are resolved that way. With the current test, the remote has `Invalid` set, the local one has not -> not equivalent, even though I believe they should be.